psydac/feec/multipatch/plotting_utilities.py

```diff
@@ -0,0 +1,300 @@
+# coding: utf-8
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+from sympy      import lambdify
+from matplotlib import cm
+from matplotlib import colors
+
+from psydac.utilities.utils    import refine_array_1d
+from psydac.fem.basic          import FemField
+from psydac.feec.pull_push     import push_2d_h1, push_2d_hcurl, push_2d_hdiv, push_2d_l2
+
+from mpi4py import MPI
+#==============================================================================
+# some plotting utilities
+# todo (MCP, april 12): merge get_grid_vals_scalar and get_grid_vals_vector into a single function
+def get_grid_vals_scalar(u, etas, mappings_list, space_kind='h1'):  #_obj):
+    # get the physical field values, given the logical field and the logical grid
+    # n_patches = len(domain)
+    n_patches = len(mappings_list)
+    u_vals    = n_patches*[None]
+    for k in range(n_patches):
+        eta_1, eta_2 = np.meshgrid(etas[k][0], etas[k][1], indexing='ij')
+        u_vals[k] = np.empty_like(eta_1)
+        if isinstance(u,FemField):
+            uk_field = u.fields[k]   # todo (MCP): try with u[k].fields?
+        else:
+            # then field is just callable
+            uk_field = u[k]
+
+        if space_kind == 'h1':
+            # todo (MCP): add 2d_hcurl_vector
+            push_field = lambda eta1, eta2: push_2d_h1(uk_field, eta1, eta2)
+        else:
+            push_field = lambda eta1, eta2: push_2d_l2(uk_field, eta1, eta2, mapping=mappings_list[k])
+        for i, x1i in enumerate(eta_1[:, 0]):
+            for j, x2j in enumerate(eta_2[0, :]):
+                u_vals[k][i, j] = push_field(x1i, x2j)
+
+    # u_vals  = np.concatenate(u_vals, axis=1)
```

Comment:
What is this commented line for? Can we remove it?

Reply:
yes, we can delete it, it's not nessary

Reply:
done in [4c54406](https://github.com/pyccel/psydac/pull/160/commits/4c5440653d00be8d85aa7c35a2b9c66927667b5a)